src/controller/CHIPDevice.h

```diff
@@ -309,6 +309,8 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
 
     void SetActive(bool active) { mActive = active; }
 
+    bool IsCSRNonceProvided() const {return mCSRNonceProvided; }
```

Comment:
where is this used?

Why is it the concern of this clas to know where CSR originates from? If we do care, it seems better to have some sort of enum:

```cpp

enum class CsrData {
   kMissing,
   kRandom,
   kApplicationProvided,
}
```

to keep track of things. Otherwise mCSRNonceProvided seems to be sticky (once you set, calls to random never make it go away) and logic seems to be a bit too implementation specificl.

So in order I would prefer 
  (a) CSR generation and knowledge how we got it should not be a concern of this class (the fact that this has a 'Generate' call seems a bit awkward to me. Too much responsibility on this class
  (b) if that is not possible (or we want to expedite code) then add an enum rather than a bool to define the CSR source. Everything that sets CSR sets the right value for what the CSR is.

Reply:
You are right, its not needed here. Removing it from this class. It's not used anywhere.